homeassistant/components/rainforest_raven/diagnostics.py

```diff
@@ -0,0 +1,45 @@
+"""Diagnostics support for a Rainforest RAVEn device."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN
+from .coordinator import RAVEnDataCoordinator
+
+TO_REDACT_CONFIG = {CONF_MAC}
+TO_REDACT_DATA = {"device_mac_id", "meter_mac_id"}
+
+
+@callback
+def async_redact_meter_macs(data: dict) -> dict:
+    """Redact meter MAC addresses from mapping keys."""
+    if not data.get("Meters"):
+        return data
+
+    redacted = {**data, "Meters": {**data["Meters"]}}
```

Comment:
This is not a very idiomatic to selectively copy the `Meters` dictionary, please consider using `copy.deepcopy` or add a comment.

The code works now, but it could break later if we change what's being mutated.

Reply:
That's fair, I added a comment in c142a16.

Reply:
I think this suggestion by @ludeeus is much clearer:

```py
    redacted = {**data, "Meters": {}}
    for idx, mac_id in enumerate(data["Meters"]):
        redacted["Meters"][f"**REDACTED{idx}**"] = data["Meters"][mac_id]
```

Reply:
Agreed, implemented in b28067d.